WooCommerce/Classes/Authentication/Navigation Exceptions/RoleErrorViewModel.swift

```diff
@@ -0,0 +1,102 @@
+import Foundation
+import UIKit
+
+final class RoleErrorViewModel {
+    /// The name of the user on the store.
+    private(set) var nameText: String
+
+    /// Assigned roles text for the user.
+    private var roles: [String] = [] {
+        didSet {
+            roleText = humanizedRolesText(from: roles)
+        }
+    }
+
+    /// Presentable version of the user's roles.
+    private(set) var roleText: String = ""
+
+    /// An illustration accompanying the error.
+    /// This is intended as a computed property to adjust to runtime color appearance changes.
+    var image: UIImage {
+        .incorrectRoleError
+    }
+
+    /// Extended description of the error.
+    let descriptionText: String = .errorMessageText
+
+    /// Provides the title for an auxiliary button
+    let auxiliaryButtonTitle: String = .linkButtonTitle
+
+    /// Provides the title for a primary action button
+    let primaryButtonTitle: String = .primaryButtonTitle
+
+    /// Provides the title for a secondary action button
+    let secondaryButtonTitle: String = .secondaryButtonTitle
+
+    /// Provides the title for the help navigation bar button
+    let helpBarButtonTitle: String = .helpBarButtonItemTitle
+
+    /// Provides the URL destination when the link button is tapped
+    private let linkDestinationURL = WooConstants.URLs.rolesAndPermissionsInfo.asURL()
+
+    /// An object capable of executing display-related tasks based on updates
+    /// from the view model.
+    weak var output: RoleErrorOutput?
+
+    // MARK: Lifecycle
+
+    init(displayName: String, roles: [String]) {
+        self.nameText = displayName
+        self.roles = roles
+
+        // Note: assignment during initialization will not trigger didSet.
+        roleText = humanizedRolesText(from: roles)
+    }
+
+    func didTapPrimaryButton() {
+        // TODO: Implement retry functionality
+    }
+
+    func didTapSecondaryButton() {
+        // TODO: Implement log out functionality
+    }
+
+    func didTapAuxiliaryButton() {
+        output?.displayWebContent(for: linkDestinationURL)
+    }
+
+    // MARK: Private Helpers
+
+    /// try to "humanize" the roles. e.g., "shop_manager" to "Shop Manager".
+    /// given multiple roles, it will all be joined with a comma separator.
+    private func humanizedRolesText(from roles: [String]) -> String {
```

Comment:
I don't see a way to get localized role descriptions from the API, but I think we should explicitly handle the core roles and localize them, specially since the other copy in the screen will localize the referenced roles.

This function might still be useful for any roles that we don't recognize

Reply:
> ... handle the core roles and localize them,
> This function might still be useful for any roles that we don't recognize

Good point! 👍 I'll make the changes on #4492 since I've refactored that function and also introduced `User+Roles` to sort of encapsulate the roles validation stuff.